languages/rust/oso/src/oso.rs

```diff
@@ -166,14 +175,24 @@ impl Oso {
         self.check_inline_queries()
     }
 
+    fn reinitialize_roles(&mut self) -> crate::Result<()> {
+        if !self.polar_roles_enabled {
+            return Ok(());
+        }
+        self.polar_roles_enabled = false;
+        self.host.del_class(OSO_INTERNAL_ROLES_HELPER);
+        self.enable_roles()
+    }
+
     /// Load a string of polar source directly.
     /// # Examples
     /// ```ignore
     /// oso.load_str("allow(a, b, c) if true;");
     /// ```
-    pub fn load_str(&self, s: &str) -> crate::Result<()> {
+    pub fn load_str(&mut self, s: &str) -> crate::Result<()> {
```

Comment:
In other languages `load_str` calls `load_file`, so the roles library is reinitialized for both.

In Rust, both are separately implemented. We'll need to either call `load_str` from `load_file` or add `reinitialize_roles` to `load_file`.

Reply:
good catch, thanks!